src/diffusers/pipelines/marigold/pipeline_marigold_depth.py

```diff
@@ -0,0 +1,661 @@
+                `latents` argument.
+
+        Examples:
+
+        Returns:
+            `MarigoldDepthOutput`: Output class instance for Marigold monocular depth prediction pipeline.
+        """
+
+        # 0. Resolving variables
+        device = self._execution_device
+        dtype = self.dtype
+
+        num_images = 1
+        if (isinstance(image, np.ndarray) or torch.is_tensor(image)) and image.ndim == 4:
+            num_images = image.shape[0]
+
+        if num_inference_steps is None:
+            num_inference_steps = self.default_denoising_steps
+        if processing_resolution is None:
+            processing_resolution = self.default_processing_resolution
+
+        # 1. Checking inputs
+        self.check_inputs(
+            image,
+            num_inference_steps,
+            ensemble_size,
+            processing_resolution,
+            resample_method_input,
+            resample_method_output,
+            batch_size,
+            ensembling_kwargs,
+            latents,
+            generator,
+            output_prediction_format,
+            output_visualization_kwargs,
+        )
+
+        # 2. Prepare empty text conditioning. Model invocation: self.tokenizer, self.text_encoder
+        if self.empty_text_embedding is None:
+            self.encode_empty_text()
+
+        # 3. Preprocessing input image
+        image, padding, original_resolution = self.image_processor.preprocess(
+            image, processing_resolution, resample_method_input, check_input, device, dtype
+        )  # [N,3,PPH,PPW]
+
+        # 4. Encode input image into latent space. Model invocation: self.vae.encoder
+        image_latent, pred_latent = self.prepare_latent(
+            image, latents, generator, ensemble_size, batch_size
+        )  # [N*E,4,h,w], [N*E,4,h,w]
+
+        del image
+
+        batch_empty_text_embedding = self.empty_text_embedding.to(device=device, dtype=dtype).repeat(
+            batch_size, 1, 1
+        )  # [B,1024,2]
+
+        # 5. Denoising loop. Model invocation: self.unet
+        clean_latent = []
+
+        with self.progress_bar(total=num_images * ensemble_size * num_inference_steps) as progress_bar:
+            for i in range(0, num_images * ensemble_size, batch_size):
+                batch_image_latent = image_latent[i : i + batch_size]  # [B,4,h,w]
+                batch_pred_latent = pred_latent[i : i + batch_size]  # [B,4,h,w]
+                B = batch_image_latent.shape[0]
+
+                batch_text_embedding = batch_empty_text_embedding[:B]  # [B,2,1024]
+
+                self.scheduler.set_timesteps(num_inference_steps, device=device)
+
+                for t in self.scheduler.timesteps:
+                    batch_latent = torch.cat([batch_image_latent, batch_pred_latent], dim=1)  # [B,8,h,w]
+                    noise = self.unet(batch_latent, t, encoder_hidden_states=batch_text_embedding).sample  # [B,4,h,w]
```

Comment:
Let's maybe do this to have better support for `torch.compile()`:

```suggestion
                    noise = self.unet(batch_latent, t, encoder_hidden_states=batch_text_embedding, return_dict=False)[0] # [B,4,h,w]
```

Reply:
+1

Reply:
Done